lib/facter/powershell_version.rb

```diff
@@ -0,0 +1,49 @@
+Facter.add(:powershell_installed) do
+ confine :operatingsystem => :windows
+ require 'win32/registry'
+ setcode do
+  begin
+    Win32::Registry::HKEY_LOCAL_MACHINE.open('SOFTWARE\Microsoft\PowerShell\1\PowerShellEngine', ::Win32::Registry::KEY_READ)
+    true
+  rescue
+    false
+  end
+ end
+end
+
+Facter.add(:powershell_version) do
+  confine :operatingsystem => :windows, :powershell_installed => true
+  require 'win32/registry'
+  setcode do
+    Win32::Registry::HKEY_LOCAL_MACHINE.open('SOFTWARE\Microsoft\PowerShell\3\PowerShellEngine') do |reg|
```

Comment:
This check may need to have a fallback to `PowerShell\1\PowerShellEngine` when the first call fails. Then maybe another fallback?